### PR TITLE
support JSON hash/array in filter_record_transformer

### PR DESCRIPTION
**Pull request by kiyoto**
a proposed implementation for #521

### CODE REVIEW

**Comment by kiyoto** [comment]
@sonots let me know if you have further feedback.

**Comment by sonots** [comment]
:+1:

File: lib/fluent/plugin/filter_record_transformer.rb [hunk 1]
```diff
@@ -92,17 +92,45 @@ def filter_stream(tag, es)
 
     private
 
+    def parse_value(value_str)
+      if /^[\[{]/.match(value_str)
```
Comment (repeatedly) [inline lib/fluent/plugin/filter_record_transformer.rb]: `start_with?('{', '[')` is more clear than `/^[\[{]/`

File: lib/fluent/plugin/filter_record_transformer.rb [hunk 2]
```diff
@@ -92,17 +92,45 @@ def filter_stream(tag, es)
...
+        JSON.parse(value_str)
+      else
+        value_str
+      end
+    rescue => e
+      log.warn "failed to parse #{value_str}", :error_class => e.class, :error => e.message
```
Comment (repeatedly) [inline lib/fluent/plugin/filter_record_transformer.rb]: This message is not good for users.
`"failed to parse #{value_str}" as json. Assume #{value_str} is string parameter` or something is good.

**Comment by repeatedly** [comment]
Please squash commit

**Comment by kiyoto** [comment]
Hm, [this broke CI](https://travis-ci.org/fluent/fluentd/builds/46952123), but the error message says the issue is in `test_out_forward.rb`. @repeatedly Any idea?

**Pull request merged**